controllers/tools/applications.js

```diff
@@ -0,0 +1,235 @@
+const express = require('express');
+const moment = require('moment');
+const concat = require('lodash/concat');
+const groupBy = require('lodash/groupBy');
+const maxBy = require('lodash/maxBy');
+const get = require('lodash/get');
+const mean = require('lodash/mean');
+const uniqBy = require('lodash/uniqBy');
+const minBy = require('lodash/minBy');
+const times = require('lodash/times');
+
+const { PendingApplication, SubmittedApplication } = require('../../db/models');
+const { getDateRange } = require('./helpers');
+
+const router = express.Router();
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+function applicationsByDay(responses) {
+    if (responses.length === 0) {
+        return [];
+    }
+
+    const grouped = groupBy(responses, function(response) {
+        return moment(response.createdAt).format(DATE_FORMAT);
+    });
+
+    const newestResponse = maxBy(responses, response => response.createdAt);
+    const oldestResponse = minBy(responses, response => response.createdAt);
+    const oldestResponseDate = moment(oldestResponse.createdAt);
+
+    const daysInRange = moment(newestResponse.createdAt)
+        .startOf('day')
+        .diff(oldestResponseDate.startOf('day'), 'days');
+
+    const dayData = times(daysInRange, function(n) {
+        const key = oldestResponseDate
+            .clone()
+            .add(n, 'days')
+            .format(DATE_FORMAT);
+
+        const responsesForDay = grouped[key] || [];
+
+        return {
+            x: key,
+            y: responsesForDay.length
+        };
+    });
+
+    return dayData;
+}
+
+function minMaxAvg(arr) {
+    const sorted = arr.slice().sort((a, b) => a - b);
+    return {
+        lowest: sorted[0] || 0,
+        highest: sorted[sorted.length - 1] || 0,
+        average: mean(sorted) || 0
+    };
+}
+
+function measureTimeTaken(data) {
+    const appDurations = data.map(row => {
+        const created = moment(row.startedAt);
+        const submitted = moment(row.createdAt);
+        return submitted.diff(created, 'days');
+    });
+    return minMaxAvg(appDurations);
+}
+
+function measureWordCounts(data) {
+    const wordCounts = data.map(
+        d =>
+            d.applicationSummary
+                .map(_ => _.value)
+                .join(' ')
+                .split(' ').length
+    );
+    return minMaxAvg(wordCounts);
+}
+
+function filterByCountry(country, appType) {
+    return function(item) {
+        if (!country) {
+            return item;
+        } else if (!item.applicationSummary && !item.applicationData) {
+            return false;
+        } else {
+            let appCountry;
+
+            if (appType === 'pending') {
+                appCountry = get(item, 'applicationData.projectCountry');
+            } else {
+                const rowCountry = item.applicationSummary.find(
+                    _ =>
+                        _.label ===
+                        'What country will your project be based in?'
+                );
```

Comment:
This is a bit ugly and makes me wonder if this field should be moved to the `applicationOverview` field so we can more easily make queries like this.

Reply:
As discussed you can do `item.applicationCountry` instead.